apps/app_audiosocket.c

```diff
@@ -108,67 +109,56 @@ static int audiosocket_exec(struct ast_channel *chan, const char *data)
 		return -1;
 	}
 
+	/* Save current channel audio format and force to linear PCM. */
+
 	writeFormat = ao2_bump(ast_channel_writeformat(chan));
 	readFormat = ao2_bump(ast_channel_readformat(chan));
 
 	if (ast_set_write_format(chan, ast_format_slin)) {
-		ast_log(LOG_ERROR, "Failed to set write format to SLINEAR for channel %s\n", chanName);
-		ao2_ref(writeFormat, -1);
-		ao2_ref(readFormat, -1);
-		close(s);
-		return -1;
+		ast_log(LOG_ERROR, "Failed to set write format to SLINEAR for channel '%s'\n", chanName);
+		res = -1;
+		goto end;
 	}
-	if (ast_set_read_format(chan, ast_format_slin)) {
-		ast_log(LOG_ERROR, "Failed to set read format to SLINEAR for channel %s\n", chanName);
 
-		/* Attempt to restore previous write format even though it is likely to
-		 * fail, since setting the read format did.
-		 */
-		if (ast_set_write_format(chan, writeFormat)) {
-			ast_log(LOG_ERROR, "Failed to restore write format for channel %s\n", chanName);
-		}
-		ao2_ref(writeFormat, -1);
-		ao2_ref(readFormat, -1);
-		close(s);
-		return -1;
+	if (ast_set_read_format(chan, ast_format_slin)) {
+		ast_log(LOG_ERROR, "Failed to set read format to SLINEAR for channel '%s'\n", chanName);
+		res = -1;
+		goto end;
 	}
 
-	res = audiosocket_run(chan, args.idStr, s);
-	/* On non-zero return, report failure */
-	if (res) {
-		/* Restore previous formats and close the connection */
-		if (ast_set_write_format(chan, writeFormat)) {
-			ast_log(LOG_ERROR, "Failed to restore write format for channel %s\n", chanName);
-		}
-		if (ast_set_read_format(chan, readFormat)) {
-			ast_log(LOG_ERROR, "Failed to restore read format for channel %s\n", chanName);
-		}
-		ao2_ref(writeFormat, -1);
-		ao2_ref(readFormat, -1);
-		close(s);
-		return res;
-	}
-	close(s);
+	pbx_builtin_setvar_helper(chan, "AUDIOSOCKET_UUID", args.idStr);
```

Comment:
These have just gotten added but I don't see their existence documented anywhere, what are they for?